### DISCUSSION TITLE
lvm2: execution failed

### DISCUSSION THREAD

**Discussion by omnigenous** [davmac314/dinit#282] [Q&A] [answered]
I'm getting this error on every boot in tty1:

```
dinit: lvm2: execution failed - setting up standard input/output descriptors: No such file or directory
```

I checked `/var/log/dinit/lvm2.log` and it is empty

I can't find anything online with same exact error, not sure if it is anything to do with dinit, let me know.

**Comment by mobin-2008** [comment]
Hi. Please post your lvm2 service description for more info.

Hint: Did you do `dinitctl reload` when your service was loaded and you changed service description after that (if it's the case)?

**Reply by omnigenous** [reply]
`/etc/dinit.d/lvm2`:
```
type      = scripted
command   = /usr/bin/vgchange --sysinit -a y
logfile   = /var/log/dinit/lvm2.log
restart   = false
after     = udev-settle
```

I didn't change anything myself.

**Comment by davmac314** [comment, marked as answer]
The likely problem is that the configured logfile directory, `/var/log/dinit/`, doesn't exist (yet) when the service starts. The service needs correct dependencies so that any required directories have been created and the root filesystem is read-only, or alternatively, it needs to use a different logging type.

This is an issue that should be reported to Artix (I'm assuming this is from Artix?), it's not an issue with Dinit itself.

For a workaround, add `log-type = buffer` and remove `logfile = /var/log/dinit/lvm2.log` from the service description file.

**Reply by omnigenous** [reply]
Thanks, [I did post on Artix forum](https://forum.artixlinux.org/index.php/topic,6390.new.html), haven't got a response just yet.

**Reply by omnigenous** [reply]
Also, meta question. Should I _close_ discussion after I choose an answer?

**Reply by davmac314** [reply]
No need to close, I think discussions can stay open indefinitely.
